microsoft/MixedRealityToolkit-Unity
# Customer request: Add support for clearing the spatial observations

This feature was requested by a customer in our online forums.

This change adds the ability to request the spatial observations be cleared (removed from the scene and freed). 

It also renames SpatialAwarenessObjParent to SpatialAwarenessObserverParent for clarity.

The Spatial Awareness demo scene has been updated to add a sphere that allows for toggling the observers (first click suspends and clears, second click resumes, etc).


Testing performed
- Build and deploy to HoloLens (il2cpp and .net backends). Confirm that the meshes clear and resume.
- Build and run in editor using an early file observer prototype.
